delira/utils/config.py

```diff
@@ -1,13 +1,509 @@
+                return super().__getitem__(int(key))
+            except (KeyError, ValueError):
+                return super().__getitem__(key)
+        else:
+            return self._traverse_keys(key.split("."), create=False)
+
+    @non_string_warning
+    def __contains__(self, key):
+        """
+        Check if key is in config
+        (also works for nested dicts with short form)
+
+        Parameters
+        ----------
+        key : str
+            key for desired value
+
+        Returns
+        -------
+        bool
+            true if key is in config
+        """
+        contain = True
+        try:
+            self[key]
+        except KeyError:
+            contain = False
+        return contain
+
+    def __str__(self):
+        """
+        String representation of config
+
+        Returns
+        -------
+        str
+            representation of config
+        """
+        return self.dumps()
+
+    def _traverse_keys(self, keys, create=False):
+        """
+        Internal helper to traverse through nested dicts
+        (iterative implementation to avoid problems with python stack)
+
+        Parameters
+        ----------
+        keys : iterable of list
+            iterable with keys which should be traversed
+        create : bool, optional
+            creates new empty configs for non existant keys, by default False
+
+        Returns
+        -------
+        Any
+            value defined by the traversed keys
+        """
+        current_level = self
+        for k in keys:
+            if k not in current_level:
+                current_level[k] = self._create_internal_dict()
+            # traverse to needed dict
+            current_level = current_level._try_key(k)
+        return current_level
+
+    def _try_key(self, key):
+        """
+        Try access with integer key first and than with a string key
+
+        Parameters
+        ----------
+        key : str or int
+            key which should be tried
+
+        Returns
+        -------
+        Any
+            value inside config
+        """
+        try:
+            return self[int(key)]
+        except (KeyError, ValueError):
+            return self[key]
+
+    def update(self, update_dict, deepcopy=False, overwrite=False):
+        """
+        Update internal dicts with dict like object
+
+        Parameters
+        ----------
+        update_dict : dictlike
+            values which should be added to config
+        deepcopy : bool, optional
+            copies values from `update_dict`, by default False
+        overwrite : bool, optional
+            overwrite existing values inside config, by default False
+
+        Raises
+        ------
+        ValueError
+            if overwrite is not enabled and `update_dict` contains same values
+            as config
+        """
+        for key, item in update_dict.items():
+            # check for overwrite
+            self._raise_overwrite(key, overwrite=overwrite)
```

Comment:
I think it would be cleaner to do this in `self._update` since we then had a function dealing with the dict as whole and mapping only a single function to all its items